vscode-dotnet-runtime-library/src/Acquisition/DotnetCoreAcquisitionWorker.ts

```diff
@@ -158,6 +203,68 @@ export class DotnetCoreAcquisitionWorker implements IDotnetCoreAcquisitionWorker
         return dotnetPath;
     }
 
+    private async checkForPartialInstalls(version : string, uninstallLocalRuntime : boolean, uninstallLocalSDK : boolean)
+    {
+        const installingVersions = this.context.extensionState.get<string[]>(this.installingVersionsKey, []);
+        const partialInstall = installingVersions.indexOf(version) >= 0;
+        if (partialInstall)
+        {
+            // Partial install, we never updated our extension to no longer be 'installing'.
+            this.context.eventStream.post(new DotnetAcquisitionPartialInstallation(version));
+
+            // Uninstall everything so we can re-install. For global installs, let the installer handle it.
+            if(uninstallLocalRuntime)
+            {
+                await this.uninstallAll();
+            }
+            else if(uninstallLocalSDK)
+            {
+                await this.uninstallRuntime(version);
+            }
+        }
+    }
+
+    private async acquireGlobalCore(globalInstallerResolver : GlobalInstallerResolver): Promise<string>
+    {
+        const installingVersion = await globalInstallerResolver.getFullVersion();
+        this.checkForPartialInstalls(installingVersion, false, false);
+
+        const installer : IGlobalInstaller = os.platform() === 'linux' ?
+            new LinuxGlobalInstaller(this.context, installingVersion) :
+            new WinMacGlobalInstaller(this.context, installingVersion, await globalInstallerResolver.getInstallerUrl());
+
+        // Indicate that we're beginning to do the install.
+        await this.addVersionToExtensionState(this.installingVersionsKey, installingVersion);
+        this.context.eventStream.post(new DotnetAcquisitionStarted(installingVersion));
+
+        // See if we should return a fake path instead of running the install
+        if(process.env.VSCODE_DOTNET_GLOBAL_INSTALL_FAKE_PATH && process.env.VSCODE_DOTNET_GLOBAL_INSTALL_FAKE_PATH === 'true')
```

Comment:
It feels really gross to have testing code in here but I see why you feel the need to do it and am struggling to think of an alternative way of having e2e test coverage.... It looks like we only have 1 e2e test that leverages this code anyways so I'm wondering if it's even worth it at all.

Reply:
Yeah, we cant do a global install on the devboxes, that would be bad. I think its worth it to have code coverage for it, otherwise we dont cover the acquisition global check whatsoever. It is not the most comprehensive test but it could catch very important 'breakages!'